h2/stream.py

```diff
@@ -701,6 +701,12 @@ def __init__(self,
         # The configuration for this stream.
         self.config = config
 
+    def __repr__(self):
+        return "<H2Stream id:%d state:%r>" % (
```

Comment:
With apologies for not spotting this sooner – hard-coding the class name in the `__repr__` means it gives misleading information when you subclass. Better to load the class name dynamically, like so:

```python
def __repr__(self):
    return "<%s id:%d state:%r>" % (
        type(self).__name__,
        self.stream_id,
        self.state_machine.state
    )
```

so if you subclass, the `repr()` picks up the name of the new class.